src/modbusrtu.cpp

```diff
@@ -4,3 +4,96 @@
  *  This source code is licensed under the MIT license found in the
  *  LICENSE file in the root directory of this source tree.
  */
+
+#include "modbusrtu.h"
+
+#include <stdio.h>
+
+// Linux headers
+#include <fcntl.h>      // Contains file controls like O_RDWR
+#include <errno.h>      // Error integer and strerror() function
+#include <termios.h>    // Contains POSIX terminal control definitions
+#include <unistd.h>     // write(), read(), close()
+
+using namespace osdev::components::modbus;
+
+ModbusRtu::ModbusRtu( const ConnectionConfig &conf )
+    : m_conConfig( conf )
+    , m_socket( -1 )
+{
+}
+
+bool ModbusRtu::Connect()
+{
+    m_socket = open( m_conConfig.getPortName().c_str(), O_RDWR);
+    if(m_socket == -1)
+    {
+        return false;
+    }
+
+    struct termios l_tty;
+
+    // Read in existing settings, and handle any error
+    if( tcgetattr(m_socket, &l_tty) != 0 )
+    {
+        // Replace later on with a logger line.
+        return false;
+    }
+
+    /* Control modes */
+    l_tty.c_cflag &= ~PARENB;                   // Clear parity bit, disabling parity (most common)
+    l_tty.c_cflag &= ~CSTOPB;                   // Clear stop field, only one stop bit used in communication (most common)
+    l_tty.c_cflag &= ~CSIZE;                    // Clear all bits that set the data field
+    l_tty.c_cflag |= CS8;                       // 8 bits per byte (most common)
```

Comment:
The comment here is misleading. `most common` is not true. For RTU, this is, by definition, 8 bits. 7 bits would be ASCII.